### PR TITLE
Product-Deploy Week 34

### LINKED ISSUE #17: Gulpfile.js should expose only the deploys
When looking in phpstorm, you can see a huge list of ALL of the gulp tasks that can be executed.

It would be better if the inner tasks are loaded later, when needed.

```
gulp.task('wk', () => {
gulp.task('assemble', done=> {...});
return runSequence('assemble');
}
```

### LINKED ISSUE #31: spici.assemble

### LINKED ISSUE #33: spici.siteDB is depricated
Has been replaced with a compiled json.

Would be good to remove the dependencies introduced by siteDB(sqlite package), to reduce the size of the package.

Might be good to be reworked in a different way, so it would validate the json's contents.

### PR DESCRIPTION
**Pull request by CZYK**
resolves #33
resolves #31
resolves #17

### CODE REVIEW

**Comment by CZYK** [comment]
@iliqportals Work in progress so don't be too harsh but feedback on the general direction is very much appreciated.

**Comment by CZYK** [comment]
@iliqportals It's not perfect but I would like to merge it into develop in this state end of the day. Please provide me your feedback so I can adapt it before merging,

File: lib/portal.js [hunk 1]
```diff
@@ -0,0 +1,255 @@
+"use strict";
+
+/**
+ * @type {Portal}
+ */
+const Portal = class Portal{
```
Comment (iliqportals) [inline lib/portal.js]: class Portal is more than sufficient
const Portal = can be removed

Comment (iliqportals) [reply lib/portal.js]: As Rob and Thijs mentioned, in order to keep everything generic : Replace the world 'portal' with 'product'
In a even more generic flow, class Portal can be named as class Deploy.
Deploy.workingCopy
Deploy.full

File: lib/portal.js [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+		this.opts = {
+			'to': '',
+			'from': process.cwd(),
+			'env': this.env.TST,
+		};
+		Object.assign(this.opts, opts);
+
+		if(!opts.to){
+
+			throw new Error(`Portal: opts.to is empty. Please provide a valid location.`);
```
Comment (iliqportals) [inline lib/portal.js]: Portal:
can be replaced with
${this.constructor.name}
This prevents inconsistancy if the class gets moved/renamed or this block of code gets reused

File: lib/portal.js [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+	 * @param {string} opts.to
+	 * @param {string} [opts.from=current working directory]
+	 * @param {string} [opts.env=Testing]
+	 */
+	constructor(opts){
+
+		let env = require('./private/env');
+
+		this.log = require('@studyportals/node-log');
+		this.env = env;
```
Comment (iliqportals) [inline lib/portal.js]: this.env is a constant encapsulator and used only in a private context.

best practice would be to put it before the class declaration

const env = require('./private/env')
class Portal{
...
}

File: lib/portal.js [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+		let valid_envs = Object.keys(env).map(function(key) {
+			return env[key];
+		});
+
+		if(valid_envs.indexOf(this.opts.env) === -1){
+
+			throw new Error(`Portal: Incorrect opts.env: '${this.opts.env}', available options: ${valid_envs.concat(', ')}`);
+		}
+
+		this.enableCompression = this.opts.env === this.env.PRD || this.opts.env === 'Live' || this.opts.env === this.env.STG;
```
Comment (iliqportals) [inline lib/portal.js]: Live should not be maintained as a possibility, can be removed

File: lib/portal.js (hunk 1 above)
Comment (CZYK) [reply lib/portal.js]: Will do!

**Pull request merged**